Building against GNU Termcap fails

**Describe the bug**
Building against [GNU Termcap][1] fails:

```
 CC     obj/x86_64-pc-linux-musl/opt/jit/prim_tty_nif.o
nifs/common/prim_tty_nif.c:46:12: fatal error: curses.h: No such file or directory
   46 |   #include <curses.h>
      |            ^~~~~~~~~~
compilation terminated.
make[4]: *** [x86_64-pc-linux-musl/Makefile:916: obj/x86_64-pc-linux-musl/opt/jit/prim_tty_nif.o] Error 1
make[4]: Leaving directory '/home/runner/build/eturnal/x64-musl/src/otp_src_26.0/erts/emulator'
make[3]: *** [/home/runner/build/eturnal/x64-musl/src/otp_src_26.0/make/run_make.mk:35: opt] Error 2
make[3]: Leaving directory '/home/runner/build/eturnal/x64-musl/src/otp_src_26.0/erts/emulator'
make[2]: *** [Makefile:45: opt] Error 2
make[2]: Leaving directory '/home/runner/build/eturnal/x64-musl/src/otp_src_26.0/erts'
make[1]: Leaving directory '/home/runner/build/eturnal/x64-musl/src/otp_src_26.0/erts'
make[1]: *** [Makefile:60: jit] Error 2
make: *** [Makefile:503: emulator] Error 2
Error: Process completed with exit code 2.
```

[1]: https://www.gnu.org/software/termutils/manual/termcap-1.3/termcap.html

**To Reproduce**
Run `./configure && make` with GNU Termcap being available instead of, e.g., ncurses.

**Expected behavior**
Building against GNU Termcap succeeds, as it did up to Erlang/OTP 25.x.

**Affected versions**
Erlang/OTP 26.0.

**Additional context**
We've linked (statically) against GNU Termcap in the past, as it's easier to build portable binaries than it is with ncurses.  However, I'm not sure whether supporting GNU Termcap is worth any hassle; or whether other non-ncurses implementations might be affected as well.  If you don't think so, just close this issue!

--- a/erts/emulator/nifs/common/prim_tty_nif.c
+++ b/erts/emulator/nifs/common/prim_tty_nif.c
@@ -43,8 +43,7 @@
 #include <locale.h>
 #ifdef HAVE_TERMCAP
 #include <termios.h>
-#include <curses.h>
-#include <term.h>
+#include <termcap.h>
 #endif
 #ifndef __WIN32__
 #include <unistd.h>
@@ -710,11 +709,12 @@ static ERL_NIF_TERM tty_tgetstr_nif(ErlNifEnv* env, int argc, const ERL_NIF_TERM
     /* tgetstr seems to use a lot of stack buffer space,
        so buff needs to be relatively "small" */
     char *str = NULL;
-    char buff[BUFSIZ] = {0};
+    char buff_area[BUFSIZ] = {0};
+    char *buff = (char*)buff_area;
 
     if (!enif_inspect_iolist_as_binary(env, argv[0], &TERM))
         return enif_make_badarg(env);
-    str = tgetstr((char*)TERM.data, (char**)&buff);
+    str = tgetstr((char*)TERM.data, &buff);
     if (!str) return atom_false;
     enif_alloc_binary(strlen(str), &ret);
     memcpy(ret.data, str, strlen(str));
@@ -742,8 +742,11 @@ static int tty_puts_putc(int c) {
 static ERL_NIF_TERM tty_tgoto_nif(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[]) {
 #ifdef HAVE_TERMCAP
     ErlNifBinary TERM;
+    ERL_NIF_TERM ret;
     char *ent;
     int value1 = 0, value2 = 0;
+    unsigned char *buff;
+
     if (!enif_inspect_iolist_as_binary(env, argv[0], &TERM) ||
         (argc > 1 && !enif_get_int(env, argv[1], &value1)) ||
         (argc > 2 && !enif_get_int(env, argv[2], &value2))
@@ -753,14 +756,12 @@ static ERL_NIF_TERM tty_tgoto_nif(ErlNifEnv* env, int argc, const ERL_NIF_TERM a
     if (!ent) return make_errno_error(env, "tgoto");
 
     tputs_buffer_index = 0;
-    if (tputs(ent, 1, tty_puts_putc)) {
-        return make_errno_error(env, "tputs");
-    } else {
-        ERL_NIF_TERM ret;
-        unsigned char *buff = enif_make_new_binary(env, tputs_buffer_index, &ret);
-        memcpy(buff, tputs_buffer, tputs_buffer_index);
-        return enif_make_tuple2(env, atom_ok, ret);
-    }
+    (void)tputs(ent, 1, tty_puts_putc); /* tputs only fails if ent is null,
+                                           which is cannot be. */
+
+    buff = enif_make_new_binary(env, tputs_buffer_index, &ret);
+    memcpy(buff, tputs_buffer, tputs_buffer_index);
+    return enif_make_tuple2(env, atom_ok, ret);
 #else
     return make_enotsup(env);
 #endif

